OpenSim/Examples/Plugins/VolumetricContactExample/VolumetricContactForce.cpp

```diff
@@ -0,0 +1,289 @@
+    //Vec3 centroidPos = eFrame.expressVectorInGround(s, Vec3(cenLx, cenLy, cenLz)); // centroid expressed in ground frame
+    Vec3 centroidPosP = eFrame.expressVectorInAnotherFrame(s, Vec3(cenLx, cenLy, cenLz), pFrame); // centroid expressed in plane frame
+
+    // NOTE: The rest of this function could be generalised for any volumetric contact.
+
+    // Relative linear and angular velocity (at centroid) expressed in plane frame
+    Vec3 v_E_cen_G = eFrame.findStationVelocityInGround(s, centroidPosE); // velocity of ellipsoid at centroid, expressed in ground frame
+    Vec3 v_P_cen_G = pFrame.findStationVelocityInGround(s, centroidPosP); // velocity of plane at centroid, expressed in ground frame
+    Vec3 v_cen_P = ground.expressVectorInAnotherFrame(s, v_E_cen_G - v_P_cen_G, pFrame); // relative velocity (of ellipsoid) at centroid, expressed in plane frame
+    Vec3 w_P = ground.expressVectorInAnotherFrame(s, // relative angular velocity, expressed in plane frame
+        eFrame.getAngularVelocityInGround(s) - pFrame.getAngularVelocityInGround(s), pFrame);
+    
+    // Normal force equation
+    double forceZP = get_k_V() * V * (1 + get_a_V() * v_cen_P[2]);
+
+    // Rolling resistance equations (moment in tangential directions)
+    double torqueXP = get_k_V() * get_a_V() * (Jp_xx * w_P[0] + Jp_xy * w_P[1]);
+    double torqueYP = get_k_V() * get_a_V() * (Jp_xy * w_P[0] + Jp_yy * w_P[1]);
+
+    // Tangential friction equations (force in tangential directions)
+    double vt_cen_P = pow(pow(v_cen_P[0], 2) + pow(v_cen_P[1], 2), 0.5); // magnitude of relative velocity at centroid
+    double forceXP, forceYP;
+    if (vt_cen_P == 0) // avoid divide by 0 errors
+    {
+        forceXP = 0;
+        forceYP = 0;
+    }
+    else
+    {
+        double mu = findMu(vt_cen_P, get_v_t());
+        forceXP = -abs(forceZP) * mu * v_cen_P[0] / vt_cen_P;
+        forceYP = -abs(forceZP) * mu * v_cen_P[1] / vt_cen_P;
+    }
+    
+    // Spinning friction (torque in normal direction)
+    double torqueZP;
+    if (V == 0) // avoid divide by 0 errors
+    {
+        torqueZP = 0;
+    }
+    else
+    {
+        double mu = findMu(w_P[2], get_w_t());
+        torqueZP = -abs(forceZP) / V * mu * (Jp_xx + Jp_yy); // note: Jp_zz = (Jp_xx + Jp_yy)
+    }
+    
+    // Convert force and torque from plane frame to ground frame
+    Vec3 forceG = pFrame.expressVectorInGround(s, Vec3(forceXP, forceYP, forceZP)); // force vector in ground frame
+    Vec3 torqueG = pFrame.expressVectorInGround(s, Vec3(torqueXP, torqueYP, torqueZP)); // torque vector in ground frame
+    
+    // Apply the forces and torques to each body:
+    applyForceToPoint(s, eFrame, centroidPosE, forceG, bodyForces);
+    applyForceToPoint(s, pFrame, centroidPosP, -forceG, bodyForces);
+    applyTorque(s, eFrame, torqueG, bodyForces);
+    applyTorque(s, pFrame, -torqueG, bodyForces);
+
+    // Debuging info
+    // --------------
+    int deb = 0;
```

Comment:
To make debugging info conditional you can use `Object::getDebugLevel()`.

Reply:
I just removed this section of code - I was not making much use of it